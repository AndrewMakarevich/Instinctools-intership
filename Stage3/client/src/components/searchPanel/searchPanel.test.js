import { render, screen } from '@testing-library/react';
import SearchPanel from './searchPanel';

describe('Correct search panel', () => {
  test('render', () => {
    render(
      <SearchPanel
        queryParams={{
          page: 1,
          limit: 2,
          filterObject: {
            groupName: '',
            groupTitle: '',
          },
        }}
        paramsMap={['groupName', 'groupTitle']}
      />
    );
    expect(screen.getByTestId('search-panel')).toBeInTheDocument();
    expect(screen.getByTestId('search-panel')).toMatchSnapshot();
  });
});
