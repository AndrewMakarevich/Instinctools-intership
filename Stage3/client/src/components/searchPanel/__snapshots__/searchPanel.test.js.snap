// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct search panel render 1`] = `
<section
  class="sc-eCYdqJ eLaAvh"
  data-testid="search-panel"
>
  <div
    class="sc-jSMfEi eeQFHx"
  >
    <label
      class="sc-gsnTZi kBVLDw"
      data-testid="search-input"
    >
      <input
        class="sc-dkzDqf dbAKRW PropTypes.string"
        id="search-input"
        placeholder="Search by groupName"
        value=""
      />
      <span
        class="sc-hKMtZM exkuXX"
      >
        <svg
          class="icon "
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </label>
    <label
      class="sc-gsnTZi kBVLDw"
      data-testid="search-input"
    >
      <input
        class="sc-dkzDqf dbAKRW PropTypes.string"
        id="search-input"
        placeholder="Search by groupTitle"
        value=""
      />
      <span
        class="sc-hKMtZM exkuXX"
      >
        <svg
          class="icon "
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </label>
  </div>
  <button
    class="sc-bczRLJ bJUkpm"
    disabled=""
    type="button"
  >
    Clear search inputs
  </button>
</section>
`;
